GameEngine/Players/EpsilonGreedyPlayer.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace GameEngine.Players
+{
+    public class EpsilonGreedyPlayer: GreedyPlayer
+    {
+        private static readonly Random Random = new Random();
+        private double Epsilon { get; }
+
+        public EpsilonGreedyPlayer(double epsilon) :
+            base()
+        {
+            Epsilon = epsilon;
+        }
+
+        public override Play FormulatePlay(GameBoard board)
+        {
+            double randomSampling = Random.NextDouble();
+
+            if (randomSampling < Epsilon)
```

Comment:
❌ These are reversed. The textbook way is to [take the random action with probability epsilon](https://junedmunshi.wordpress.com/2012/03/30/how-to-implement-epsilon-greedy-strategy-policy/).